message.go

```diff
@@ -52,6 +55,9 @@ func (m *Message) encode(pe packetEncoder) error {
 	pe.putInt8(m.Version)
 
 	attributes := int8(m.Codec) & compressionCodecMask
+	if m.LogAppendTime {
+		attributes |= timestampTypeMask
+	}
```

Comment:
Does this make sense to be set by a producer? I thought this was a server config setting.